py/tests/ctm_csc.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/python3
+
+import pykms
+
+def ctm_to_blob(ctm, card):
+    len=9
+    arr = bytearray(len*8)
+    view = memoryview(arr).cast("i")
+
+    for x in range(len):
+        i, d = divmod(ctm[x], 1)
+        view[x * 2 + 0] = int(d * ((2 ** 31) - 1))
+        view[x * 2 + 1] = int(i)
+        #print("%f = %x.%08x" % (ctm[x], view[x * 2 + 1], view[x * 2 + 0]))
+
+    return pykms.Blob(card, arr);
+
+
+card = pykms.Card()
+res = pykms.ResourceManager(card)
+conn = res.reserve_connector("unknown")
+crtc = res.reserve_crtc(conn)
+mode = conn.get_default_mode()
+
+fb = pykms.DumbFramebuffer(card, mode.hdisplay, mode.vdisplay, "XR24");
+pykms.draw_test_pattern(fb);
+
+crtc.set_mode(conn, fb, mode)
+
+input("press enter to set normal ctm\n")
+
+ctm = [ 1.0,	0.0,	0.0,
+        0.0,	1.0,	0.0,
+        0.0,	0.0,	1.0 ]
+
+ctmb = ctm_to_blob(ctm, card)
+
+crtc.set_prop("CTM", ctmb.id)
+
+input("press enter to set r->g g->b b->r ctm\n")
+
+ctm = [ 0.0,	1.0,	0.0,
+        0.0,	0.0,	1.0,
+        1.0,	0.0,	0.0 ]
+
+ctmb = ctm_to_blob(ctm, card)
```

Comment:
But the previous blob has been committed already. Surely, and also according to result of my brief test [1], kernel side keeps a reference of blobs that are part of the current state. I would call any other behaviour a bug.

I added a following sequence:
-----------------------------------------------------------------
ctm = [ 0.0,	1.0,	0.0,
        0.0,	0.0,	1.0,
        1.0,	0.0,	0.0 ]

ctmb = ctm_to_blob(ctm, card)

crtc.set_prop("CTM", ctmb.id)

input("test")

ctmb = 0

input("test")

crtc.set_mode(conn, fb, mode)

input("test")
-----------------------------------------------------------------
... and the committed color matrix remains active also after setting the crtc mode.